src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -1148,81 +1079,53 @@ public void saveAimlFile(String botName, String filename, String data) {
 
   @Override
   public ServiceConfig getConfig() {
-    ProgramABConfig config = (ProgramABConfig) super.getConfig();
-    // REMOVED from overlap with subscriptions
-    // Set<String> listeners = getAttached("publishText");
-    // config.textListeners = listeners.toArray(new String[listeners.size()]);
-
-    // listeners = getAttached("publishUtterance");
-    // config.utteranceListeners = listeners.toArray(new
-    // String[listeners.size()]);
-    if (config.bots == null) {
-      config.bots = new ArrayList<>();
+    ProgramABConfig c = (ProgramABConfig) super.getConfig();
+    if (c.bots == null) {
+      c.bots = new ArrayList<>();
     }
 
-    config.bots.clear();
+    c.bots.clear();
     for (BotInfo bot : bots.values()) {
 
       Path pathAbsolute = Paths.get(bot.path.getAbsolutePath());
       Path pathBase = Paths.get(System.getProperty("user.dir"));
       Path pathRelative = pathBase.relativize(pathAbsolute);
-      config.bots.add(pathRelative.toString());
+      c.bots.add(pathRelative.toString());
 
     }
 
-    config.currentBotName = currentBotName;
-    config.currentUserName = currentUserName;
-
-    return config;
+    return c;
   }
 
   @Override
-  public ServiceConfig apply(ServiceConfig c) {
-    ProgramABConfig config = (ProgramABConfig) super.apply(c);
-    if (config.bots != null && config.bots.size() > 0) {
-      bots.clear();
-      for (String botPath : config.bots) {
+  public ServiceConfig apply(ServiceConfig config) {
+    ProgramABConfig c = (ProgramABConfig) super.apply(config);
+    if (c.bots != null && c.bots.size() > 0) {
+      // bots.clear();
+      for (String botPath : c.bots) {
         addBotPath(botPath);
       }
     }
     
-    if (config.botDir == null) {
-      config.botDir = getResourceDir();
+    if (c.botDir == null) {
+      c.botDir = getResourceDir();
     }
 
-    List<File> botsFromScanning = scanForBots(config.botDir);
+    List<File> botsFromScanning = scanForBots(c.botDir);
     for (File file : botsFromScanning) {
       addBotPath(file.getAbsolutePath());
     }
 
-    if (config.currentBotName != null) {

```

Comment:
Why's this removed? Is the bot name deprecated?

Reply:
good catch - fixed ... I thought initially it wouldn't make a difference, now that config is normalized, but the set functions broadcast state too .. so yeah, it should do it as it did before